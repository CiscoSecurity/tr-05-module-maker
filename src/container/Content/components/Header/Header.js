import React from "react";
import './Header.scss';
import * as Constants from "globals/constants/constants";
import Icons from "globals/icons/sprite.svg";


class Header extends React.Component {
    render() {
        return (
            <div className="header">
                <div>
                    <svg className="iconCiscoLogo">
                        <use xlinkHref={`${Icons}#icon-cisco-logo`}/>
                    </svg>
                </div>
                <div>
                    <p className="headerTitle">{Constants.HEADER_TITLE}</p>
                </div>
            </div>
        )
    }
}

export default Header;
