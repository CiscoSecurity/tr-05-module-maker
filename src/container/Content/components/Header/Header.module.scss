@import "src/globals/constants/colors";

.header {
    width: 100%;
    height: 49px;
    position: fixed;
    z-index: 1;
    padding-left: 20px;
    padding-right: 20px;
    border-bottom: 1px solid $border-color;
    background-color: $white-color;
    display: flex;
    align-items: center;
    overflow: visible;
}

.iconCiscoLogo {
    fill: $blue-background;
    width: 38px;
    height: 20px;
    margin-right: 20px;
}

.title {
    white-space: nowrap
}