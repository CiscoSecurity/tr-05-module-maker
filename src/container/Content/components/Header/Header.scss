@import "src/globals/constants/colors";

.header {
  width: 100%;
  height: 49px;
  position: fixed;
  padding-left: 20px;
  padding-right: 20px;
  border-bottom: 1px solid $border-color;
  background-color: $white-color;
  display: flex;
  align-items: center;
  overflow: visible;
  z-index: 2;

  .iconCiscoLogo {
    fill: $blue-background;
    width: 38px;
    height: 20px;
    margin-right: 20px;
  }

  .headerTitle {
    white-space: nowrap;
    margin: 0;
  }
}
