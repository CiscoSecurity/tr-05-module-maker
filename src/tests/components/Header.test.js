import { shallow } from 'enzyme';
import Header from "../../container/Content/components/Header/Header";
import React from "react";
import toJson from 'enzyme-to-json';


describe('Header', () => {
    it('should render a header', () => {
        const wrapper = shallow(<Header/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
