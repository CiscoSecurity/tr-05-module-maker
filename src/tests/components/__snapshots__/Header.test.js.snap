// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render a header 1`] = `
<div
  className="header"
>
  <div>
    <svg
      className="iconCiscoLogo"
    >
      <use
        xlinkHref="sprite.svg#icon-cisco-logo"
      />
    </svg>
  </div>
  <div>
    <p
      className="headerTitle"
    >
      Threat Response Module Maker
    </p>
  </div>
</div>
`;
